app/sample_manifest_excel/sample_manifest_excel/worksheet/data_worksheet.rb

```diff
@@ -83,9 +83,16 @@ def add_columns
       def create_row(detail)
         axlsx_worksheet.add_row do |row|
           columns.each do |column|
-            style = find_or_create_style(column.style)&.reference
-
-            row.add_cell column.attribute_value(detail), type: column.type, style: style
+            # If the row is invalid in the manifest, the cell should be locked
+            if sample_manifest.invalid_wells&.include?(detail.fetch(:position, nil))
+              row.add_cell column.attribute_value(detail),
+                           type: column.type,
+                           style: find_or_create_style([:disabled])&.reference
+            else
+              style = find_or_create_style(column.style)&.reference
+
+              row.add_cell column.attribute_value(detail), type: column.type, style: style
+            end
```

Comment:
Just a small refactoring to avoid a bit of duplication (haven't tested):

```
columns.each do |column|
  # If the row is invalid in the manifest, the cell should be locked
  if sample_manifest.invalid_wells&.include?(detail.fetch(:position, nil))
    style_name = [:disabled]
  else
    style_name = column.style
  end

  style = find_or_create_style(column.style)&.reference
  
  row.add_cell column.attribute_value(detail), type: column.type, style: style
end
```

Reply:
Yeah thats cleaner - I have pushed a change along the same lines.